pkg/sql/plan/query_builder.go

```diff
@@ -91,10 +91,11 @@ func (builder *QueryBuilder) remapAllColRefs(nodeID int32, colRefCnt map[[2]int3
 
 		tag := node.BindingTags[0]
 		newTableDef := &plan.TableDef{
-			Name:          node.TableDef.Name,
-			Defs:          node.TableDef.Defs,
-			Name2ColIndex: node.TableDef.Name2ColIndex,
-			Createsql:     node.TableDef.Createsql,
+			Name:               node.TableDef.Name,
+			Defs:               node.TableDef.Defs,
+			Name2ColIndex:      node.TableDef.Name2ColIndex,
+			Createsql:          node.TableDef.Createsql,
+			TableFunctionParam: node.TableDef.TableFunctionParam,
```

Comment:
add TableFunctionParam to DeepCopyTableDef.
Sometime we will reuse plan(eg: prepare)

Reply:
Ok